main.js

```diff
@@ -0,0 +1,75 @@
+function getRandomInt(min, max) {
+	return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+function randomStep() {
+	const zeroLocation = matrix.findLocation(0);
+	const randomElement = matrix.getRandomStep();
+	if (matrix.exchange(randomElement, zeroLocation))
+		generateTable(matrix);
+}
+
+function locationOffset(location, x, y) {
+	return {
+		x: location.x + x,
+		y: location.y + y
+	}
+}
+
+function onItemClick(e) {
+	const value = parseInt(e.currentTarget.innerText);
+	if (value === 0)
+		return;
+
+	const clickedLocation = matrix.findLocation(value);
+	const emptyLocation = matrix.findLocation(0);
+
+	if (matrix.exchange(clickedLocation, emptyLocation))
+		generateTable(matrix);
+
+	if (matrix.checkWin())
+		setTimeout(function () {
+			alert("Congratulations!!!");
+		}, 0);
+}
+
+function onShuffleClick() {
+	let i = 0;
+	const RenderTimeoutInterval = 30;
+	const ShufflesCount = 3;
+	const intervalId = setInterval(function () {
+		i++;
+		randomStep();
+		if (i === ShufflesCount) {
+			clearInterval(intervalId);
+		}
+	}, RenderTimeoutInterval);
+}
+
+function generateTable() {
```

Comment:
do you generate the table after each step?

Reply:
yes. I know it's bad practice. It's been implemented as a quick solution.

Reply:
it's inappropriate practive. moreover where do you remove the previous table?